### PR TITLE
modernize hs.wifi and hs.wifi.watcher

### PR DESCRIPTION
**Pull request by asmagill**
Removes methods deprecated in 10.10

Adds watchers for link quality, BSSID, power, etc. in addition to default of SSID change.

Addresses #1131

### CODE REVIEW

File: extensions/wifi/internal.m [hunk 1]
```diff
@@ -110,14 +58,14 @@ - (void)invokeCallback:(id)object {
         lua_State *L = [skin L] ;
         [[LuaSkin shared] pushLuaRef:refTable ref:_fnRef];
         if ([object isKindOfClass:[NSError class]]) {
-            log_to_console(L, _cINFO, [(NSError *)object localizedDescription]) ;
+            [LuaSkin logInfo:[(NSError *)object localizedDescription]] ;
```
Comment (cmsj) [inline extensions/wifi/internal.m]: This method looks like it has a lot of built-up LuaSkin weirdness. We have a `skin` object at the top, so the second `LuaSkin shared` is unnecessary, and these log calls could use the instance variants too.

**Comment by asmagill** [comment]
holdover from copy/paste after pulling the outdated logging code... it's cleaner now :-)

**Comment by asmagill** [comment]
@cmsj, noticed that your latest comment was tagged as a review, not a comment... How does that work?  Was the fact that you made a line-tem note sufficient to make it a review, or is their a field/toggle somewhere that I'm missing?

**Comment by asmagill** [comment]
@cmsj, the travis log shows (among other things):

~~~
/Users/travis/.rvm/gems/ruby-2.0.0-p648/gems/xcpretty-0.2.3/lib/xcpretty/syntax.rb:24:in `highlight_with_formatter': undefined method `include?' for nil:NilClass (NoMethodError)
	from /Users/travis/.rvm/gems/ruby-2.0.0-p648/gems/xcpretty-0.2.3/lib/xcpretty/syntax.rb:15:in `highlight'
	from /Users/travis/.rvm/gems/ruby-2.0.0-p648/gems/xcpretty-0.2.3/lib/xcpretty/formatters/formatter.rb:171:in `format_failure'
...
~~~

Since it seems to be a ruby error, I'm uncertain... I can merge this locally with master and it builds fine, both App and documentation, so I don't know what to do next...

Is this because of changes you're making so Travis can support the documentation generation changes?

**Comment by cmsj** [comment]
Shouldn't be anything to do with my docs work, none of that has hit master yet.

The error seems to be in the xcpretty stuff that we use in travis to make build logs not overflow the Travis limits. Perhaps we should have pinned that to a known working version. I'll try to investigate later today and when it's fixed I'll land this. 

As for the review thing - yes, starting a line comment these days starts a review.

**Pull request merged**